crates/rome_js_analyze/src/analyzers/nursery/no_misused_new.rs

```diff
@@ -0,0 +1,232 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::{markup, MarkupBuf};
+use rome_js_syntax::{
+    AnyTsTypeMember, JsClassDeclaration, JsLanguage, TsDeclareStatement, TsInterfaceDeclaration,
+    TsReferenceType, TsTypeAliasDeclaration,
+};
+use rome_rowan::{declare_node_union, AstNode, SyntaxToken, TextRange};
+
+declare_rule! {
+    /// Enforce proper usage of `new` and `constructor`.
+    ///
+    /// In JavaScript, classes utilize the `constructor` method to initialize a new instance. On the other hand, TypeScript interfaces can describe a class type with a `new()` method signature, though this pattern is not commonly seen in real-world code. Developers, especially those new to JavaScript or TypeScript, might occasionally confuse the use of `constructor` with `new`.
+    /// This rule triggers warnings in the following scenarios:
+    /// - When a class has a method named `new`.
+    /// - When an interface defines a method named `constructor` or `new` that returns the interface type.
+    /// - When a type alias has a `constructor` method.
+    ///
+    /// You should not use this rule if you intentionally want a class with a `new` method, and you're confident nobody working in your code will mistake it with an `constructor`.
+    ///
+    /// Source: https://typescript-eslint.io/rules/no-misused-new/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// declare class C {
+    ///   new(): C;
+    /// }
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// interface I {
+    ///   new (): I;
+    ///   constructor(): void;
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```ts
+    /// declare class C {
+    ///   constructor();
+    /// }
+    ///
+    /// interface I {
+    ///   new (): C;
+    /// }
+    /// ```
+    ///
+    pub(crate) NoMisusedNew {
+        version: "next",
+        name: "noMisusedNew",
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub(crate) DeclarationQuery = TsInterfaceDeclaration |  TsTypeAliasDeclaration | JsClassDeclaration | TsDeclareStatement
```

Comment:
ARe there some advantages to query on these types instead of querying `AnyTsTypeMember` / `AnyJsClassMember`?

Reply:
To know class and interface identifier, I think we need to hit the node types.
for example, 
```
interface I { // we can access this `I` through the TsInterfaceDeclaration
  new (): I;
}
```
so I set those node types.